gobblin-service/src/main/java/org/apache/gobblin/service/modules/dataset/DatasetDescriptor.java

```diff
@@ -41,4 +65,15 @@
    * This check is non-commutative.
    */
   public boolean isCompatibleWith(DatasetDescriptor other);
+
+  /**
+   * @return true if the platform-independent properties of this {@link DatasetDescriptor} are compatible with the
+   * platform-independent properties of the other {@link DatasetDescriptor}.
+   */
+  public boolean isPropertyCompatibleWith(DatasetDescriptor other);
```

Comment:
What are `platform-independent` properties? Can we group them similar to `EncryptionConfig`? One can specifically get the group and check its compatibility instead of the general `DatasetDescriptor` provides a partial and unclear compatibility check besides a full compatibility check.

Reply:
By platform-independent properties, I really mean the dataset format which includes: 1. the data representation (e.g. avro/json/csv), 2. the encoding (e.g. gzip/bzip2/deflate), and 3. encryption config. Definitely possible to group them like EncryptionConfig.